src/corecel/cont/Span.hh

```diff
@@ -251,10 +250,17 @@ make_array(Span<T, N> const& s)
 
 //---------------------------------------------------------------------------//
 //! Construct an array from a fixed-size span, removing LdgValue marker
-template<class T, std::size_t N>
-CELER_CONSTEXPR_FUNCTION Array<T, N> make_array(LdgSpan<T, N> const& s)
+template<class T,
+         std::size_t N,
+         class T2 = std::remove_cv_t<typename LdgValue<T>::value_type>>
+CELER_CONSTEXPR_FUNCTION Array<T2, N> make_array(LdgSpan<T, N> const& s)
 {
-    return make_array<typename LdgValue<T>::value_type, N>(s);
+    Array<T2, N> result{};
+    for (std::size_t i = 0; i < N; ++i)
+    {
+        result[i] = s[i];
+    }
+    return result;
```

Comment:
We need to document why the code is being duplicated. Also is it possible there's an issue in the `template<class U> Span(Span<U>)` constructor?

Reply:
that constructor looks correct to me. If we take the scenarios of constructing a `Span<T const>` from `LdgSpan<T const` which is what we were doing before this PR: `data()` casts a `LdgIterator<T consts>` to `T const*` which is used to initialize the pointer of the new span which is the same type as the cast expression.

Reply:
OK as long as it's not accidentally `reinterpret_cast`ing ;)